WcaOnRails/app/controllers/api/v0/api_controller.rb

```diff
@@ -112,6 +112,37 @@ def delegates
     paginate json: User.delegates
   end
 
+  def records
+    concise_results_date = Timestamp.find_by(name: "compute_auxiliary_data_end").date
+    cache_key = "records/#{concise_results_date.iso8601}"
+    json = Rails.cache.fetch(cache_key) do
+      records = ActiveRecord::Base.connection.exec_query <<-SQL
+        SELECT 'single' type, MIN(best) value, countryId country_id, eventId event_id
+        FROM ConciseSingleResults
+        GROUP BY countryId, eventId
+        UNION ALL
+        SELECT 'average' type, MIN(average) value, countryId country_id, eventId event_id
+        FROM ConciseAverageResults
+        GROUP BY countryId, eventId
+      SQL
+      records = records.to_hash
+      {
+        world_records: records_by_event(records),
+        continental_records: records.group_by { |record| Country.c_find(record["country_id"]).continentId }.transform_values!(&method(:records_by_event)),
+        national_records: records.group_by { |record| record["country_id"] }.transform_values!(&method(:records_by_event)),
+      }
+    end
+    render json: json
+  end
+
+  private def records_by_event(records)
+    records.group_by { |record| record["event_id"] }.transform_values! do |event_records|
+      event_records.group_by { |record| record["type"] }.transform_values! do |type_records|
+        type_records.map { |record| record["value"] }.min
```

Comment:
Do we have to worry at all about DNF/DNS here? Their values are negative, so usually the min.

Reply:
`Concise*Results` tables include only successful attempts, which is ensured by this line https://github.com/thewca/worldcubeassociation.org/blob/4449cf19bf91f1cd950f9f11b375ecaf05e8711b/WcaOnRails/lib/auxiliary_data_computation.rb#L54